core/src/main/java/org/mapfish/print/test/util/ImageSimilarity.java

```diff
@@ -356,11 +354,15 @@ public void assertSimilarity(final File actualFile, final double maxDistance) th
    */
   public void assertSimilarity(final BufferedImage actualImage, final double maxDistance)
       throws IOException {
-    if (!this.expectedPath.exists()) {
+    if (REGENERATE_EXPECTED_IMAGES || !this.expectedPath.exists()) {
       ImageIO.write(actualImage, "png", expectedPath);
-      throw new AssertionError(
-          "The expected file was missing and has been generated: "
-              + expectedPath.getAbsolutePath());
+      if (REGENERATE_EXPECTED_IMAGES) {
+        return;
```

Comment:
if (REGENERATE_EXPECTED_IMAGES == false) {
Then no need of lines 360 and 361

Reply:
Yes, I need them also the lines 367 and next will run...